base/cvd/cuttlefish/host/libs/xdg/directories.cpp

```diff
@@ -0,0 +1,158 @@
+//
+// Copyright (C) 2024 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "host/libs/xdg/directories.h"
+
+#include <dirent.h>
+
+#include <string>
+
+#include <android-base/file.h>
+#include <android-base/strings.h>
+
+#include "common/libs/utils/environment.h"
+#include "common/libs/utils/files.h"
+#include "common/libs/utils/result.h"
+#include "common/libs/utils/users.h"
+
+// See https://specifications.freedesktop.org/basedir-spec/latest/
+
+namespace cuttlefish {
+namespace {
+
+std::optional<std::string> NonEmptyEnv(const std::string& var_name) {
+  std::optional<std::string> opt = StringFromEnv(var_name);
+  return opt.has_value() && *opt == "" ? opt : std::optional<std::string>();
+}
+
+std::string Home() {
+  std::string fallback = NonEmptyEnv("HOME").value_or(CurrentDirectory());
```

Comment:
I'd prefer to fail altogether than to fallback to the current directory. The current directory might be one accessible to everyone in the machine and not just the current user, not a safe place to store credentials.

Reply:
Done